WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift

```diff
@@ -132,9 +136,17 @@ final class TotalsViewModel: ObservableObject, TotalsViewModelProtocol {
     /// Called when the onboarding UI is dismissed.
     /// It is set when receiving an onboarding view model to display, and necessary to reset the internal onboarding state on UI dismissal.
     func cancelOnboarding() {
+        if let onboardingViewModel = cardPresentPaymentOnboardingViewModel {
```

Comment:
At least sometimes, `cardPresentPaymentOnboardingViewModel` can be set to `nil` before this is called.

Reply:
Thanks a lot for catching this and looking into the reason! 🙏 

I made some changes in https://github.com/woocommerce/woocommerce-ios/pull/14284/commits/7f79802b62d99779fb3fbc85c5f40b4727c8875a:

There are two ways to dismiss the onboarding with the POS modal:
- External dismissal: merchant taps X or Contact Support
- Internal dismissal: merchant taps outside of the modal

For external dismissal, `cardPresentPaymentOnboardingViewModel` has to be set to nil to dismiss the onboarding in SwiftUI. I updated the `TotalsViewModel.cancelOnboarding` to guard on the onboarding view model then perform the necessary reset and event tracking. The downside is that for external dismissal, this method is now called twice (no-op the second time) because the `posModal`'s dismiss closure is triggered for either dismissal. I added a comment about this to the method, but lemme know if you think there's a better way to handle both dismissal types!

Reply:
Thanks for the fix! I think the fact that it's called twice is not ideal but fine, I don't see any side effect of it 👍